lib/cli.js

```diff
@@ -63,6 +63,9 @@ function CLI(params) {
 	// call the Hook constructor
 	Hook.call(this);
 
+	// scan for local global hooks
+	this.scanHooks(path.join(process.cwd(), 'plugins'), /titanium-global-plugin\.js$/);
```

Comment:
You should not reference `cwd`. Titanium CLI does not make any assumptions about `cwd` and allows you to explicitly define paths such as the project directory. For example, you could be in Project A, but build Project B and it would load the plugin from Project A. That's a no go.

You could ensure that the `cwd` is a project directory by checking for a `tiapp.xml`, but we already load plugins for a project directory: https://github.com/appcelerator/titanium_mobile/blob/a0224781441a8c13b906f65dbd7236456b87a399/cli/commands/build.js#L172.

If you need to hook into an earlier hook or non-build command (which non-build hooks are deprecated), you can always override the config: `ti build --config '{"paths":{"hooks":["plugins"]}}'`.

Also, only under certain circumstances would we accept a hard-coded filename like this. This complicates backwards compatibility when we do major changes to things like the hook system, which we are going to do.

Reply:
This would allow you to do things like replace the `tiapp.xml` with a JSON based config.  Currently, if the `tiapp.xml` file is missing, everything goes bonkers...

Reply:
But we're not going to do that.

Reply:
I know.  But this would allow developers in the community to do it.  The more hooks and extensibility points a product has, the better.   Priorities don't always allow us to add every feature but if we can free the community to add their own features through hooks and other extensibility points, the product (and community) can flourish!

Reply:
Here's an idea. How about you have `titanium-global-plugin.js` in a **global** directory and that **global** directory is added to the **global** "paths.hooks" config setting, then have `titanium-global-plugin.js` call `cli.scanHooks()` in `init()` with the `cwd` or whatever you want.

Reply:
That is a whole lot more that has to get installed/loaded/configured every time we do a cloud build (GitHub Actions, Azure DevOps, etc.).  My goal is to make this easy and simple to use.   In your world, if I share a project with someone, it will fail to build and get errors because the person I shared with doesn't happen to have that global plugin installed.     I want this to be easy for developers.  Ideally, any developer could clone a Titanium app repo on GitHub and run the app in a simulator without messy workarounds and checking for pre-existing conditions.

Reply:
Then they should refrain from using unsanctioned configuration files.